Lib/functools.py

```diff
@@ -885,15 +885,7 @@ def _find_impl(cls, registry):
             match = t
     return registry.get(match)
 
-def singledispatch(func):
-    """Single-dispatch generic function decorator.
-
-    Transforms a function into a generic function, which can have different
-    behaviours depending upon the type of its first argument. The decorated
-    function acts as the default implementation, and additional
-    implementations can be registered using the register() attribute of the
-    generic function.
-    """
+def _singledispatchimpl(func, is_method):
```

Comment:
I tend to mark boolean parameters as kw-only:
```suggestion
def _singledispatchimpl(func, *, is_method):
```